glean-core/android/src/main/java/mozilla/telemetry/glean/Glean.kt

```diff
@@ -38,12 +40,14 @@ data class BuildInfo(val versionCode: String, val versionName: String, val build
 
 internal class OnGleanEventsImpl(val glean: GleanInternalAPI) : OnGleanEvents {
     override fun initializeFinished() {
-        // At this point, all metrics and events can be recorded.
-        // This should only be called from the main thread. This is enforced by
-        // the @MainThread decorator and the `assertOnUiThread` call.
-        MainScope().launch {
-            ProcessLifecycleOwner.get().lifecycle.addObserver(glean.gleanLifecycleObserver)
+        // Only set up the lifecycle observers if we don't provide a custom
+        // data path.
+        if (glean.getIsCustomDataPath() != null && glean.getIsCustomDataPath() == false) {
+            MainScope().launch {
+                ProcessLifecycleOwner.get().lifecycle.addObserver(glean.gleanLifecycleObserver)
+            }
         }
```

Comment:
Maybe a less java-like and more kotlin-like way to write this would be something like:
```
glean.getIsCustomDataPath()?.takeIf { !it }?.let {
	MainScope().launch {
		ProcessLifecycleOwner.get().lifecycle.addObserver(glean.gleanLifecycleObserver)
	}
}
```

the `?` operator will handle the null for you and the `takeIf { !it }` will process on the returned value and return `null` if `getIsCustomDataPath` returns true, so the second `?` would escape. Otherwise, if it's not `null` and it returns `false` (`!it`), then it falls into the `let { ... }` block.

Checking for `null` in Kotlin is just sorta weird, but feel free to find something that is more readable if my suggestion is too arcane or something :)

Reply:
`getIsCustomDataPath` should just be `isCustomDataPath`.

Reply:
I changed this to just use `isCustomDataPath` now and it will never be null, so that should simplify the check. I appreciate the Kotlin way to do it though, I'll keep that in mind for the future!

https://github.com/mozilla/glean/pull/2433/commits/6fbe824df65632daecdba0d917d42b38d6254c76#diff-aa8825c678ddc5d873d6b93e255f5b246fe265822b922a6f50d775b0e98b1e97R45